Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -40,68 +40,81 @@ def run
         end
         puts cask_upgrades.join(", ")
 
+        caught_exceptions = []
         outdated_casks.each do |old_cask|
-          odebug "Started upgrade process for Cask #{old_cask}"
-          raise CaskUnavailableError.new(old_cask, "The Caskfile is missing!") if old_cask.installed_caskfile.nil?
+          begin
+            upgrade_cask(old_cask)
+          rescue CaskError => e
+            caught_exceptions << e
+            next
+          end
+        end
+        return if caught_exceptions.empty?
```

Comment:
```ruby
if caught_exceptions.length > 1
  ...
elsif caught_exceptions.length == 1
  ...
end
``` 

might be a bit more readable?

Reply:
I agree. I originally had something like:
```
return if caught_exceptions.empty?
if caught_exceptions.length > 1
  raise MultipleCaskErrors, caught_exceptions
elsif caught_exceptions.length == 1
  raise caught_exceptions[0]
end
```
But `brew style` complained with the message `Use a guard clause instead of wrapping the code inside a conditional expression.`

It seems like the linter wants all single line if statements to be guard clauses?

Reply:
Add `# rubocop:disable Style/GuardClause` to this bit and it should go away.

Reply:
Thank you!

Reply:
Don't disable RuboCop for this. Just move the `if` after the `raise`.

Reply:
If I change the `if` back into a guard clause, then I'll have to change the `elsif` into a guard clause as well, and then the code will be the same as it was before @MikeMcQuaid requested the change.

Reply:
@zachauten That's fine; I defer to RuboCop 😁

Reply:
We don't get notified on force pushes so give me a shout when you've done that!

Reply:
Thanks, I didn't know that about force pushing.